core/src/main/scala/kafka/admin/ReassignPartitionsCommand.scala

```diff
@@ -79,11 +79,13 @@ object ReassignPartitionsCommand extends Logging {
     var changed = false
 
     //If all partitions have completed remove the throttle
-    if (reassignedPartitionsStatus.forall { case (topicPartition, status) => status == ReassignmentCompleted }) {
+    if (reassignedPartitionsStatus.forall { case (_, status) => status == ReassignmentCompleted }) {
       //Remove the throttle limit from all brokers in the cluster
+      //(as we no longer know which specific brokers were involved in the move)
       for (brokerId <- zkUtils.getAllBrokersInCluster().map(_.id)) {
         val configs = AdminUtils.fetchEntityConfig(zkUtils, ConfigType.Broker, brokerId.toString)
-        if (configs.remove(DynamicConfig.Broker.ThrottledReplicationRateLimitProp) != null){
+        if (configs.remove(DynamicConfig.Broker.ThrottledLeaderReplicationRateProp) != null
+          || configs.remove(DynamicConfig.Broker.ThrottledFollowerReplicationRateProp) != null){
```

Comment:
Hmm, if the first test returns true, the code for removing follower rate won't be called?